packages/shared/lib/services/sync/run.service.ts

```diff
@@ -389,6 +385,11 @@ export default class SyncRun {
                 const errorType = this.determineErrorType();
 
                 return { success: false, error: new NangoError(errorType, errorMessage), response: result };
+            } finally {
+                if (!this.isAction) {
```

Comment:
This should actually check `!isInvokedImmediately`

```
this.isInvokedImmediately = Boolean(config.isAction || config.isWebhook);
```

which basically means it is a sync

Reply:
good call 👍🏻